packages/core/contracts/financial-templates/perpetual-multiparty/PositionManager.sol

```diff
@@ -496,39 +474,48 @@ contract PositionManager is FeePayer, AdministrateeInterface {
         tokenCurrency.burn(numTokens.rawValue);
     }
 
+    function repay(FixedPoint.Unsigned memory numTokens)
+        public
+        notEmergencyShutdown()
+        noPendingWithdrawal(msg.sender)
+        fees()
+        nonReentrant()
+        returns (FixedPoint.Unsigned memory amountRepaired)
+    {
+        // TODO: implementation
+    }
+
     /**
-     * @notice After a contract has passed expiry all token holders can redeem their tokens for underlying at the
-     * prevailing price defined by the DVM from the `expire` function.
+     * @notice If the contract is emergency shutdown then all token holders and sponsors can redeem their tokens or
+     * remaining collateral for underlying at the prevailing price defined by a DVM vote.
      * @dev This burns all tokens from the caller of `tokenCurrency` and sends back the proportional amount of
      * `collateralCurrency`. Might not redeem the full proportional amount of collateral in order to account for
      * precision loss. This contract must be approved to spend `tokenCurrency` at least up to the caller's full balance.
      * @return amountWithdrawn The actual amount of collateral withdrawn.
      */
-    function settleExpired()
+    function settleEmergencyShutdown()
```

Comment:
`settleExpire`'s logic is exactly as what is needed for `settleEmergencyShutdown`.